pypesto/objective/amici_objective.py

```diff
@@ -162,20 +179,18 @@ def call_amici(
                     snllh -= rdata['sllh']
                     # TODO: Compute the full Hessian, and check here
                     ssnllh -= rdata['FIM']
+
             elif mode == Objective.MODE_RES:
                 res = np.hstack([res, rdata['res']]) \
                     if res.size else rdata['res']
                 if sensi_order > 0:
                     sres = np.vstack([sres, rdata['sres']]) \
                         if sres.size else rdata['sres']
 
-        # map_to_output is called twice, might be prettified
-        return Objective.map_to_output(
-            sensi_orders,
-            mode,
-            fval=nllh, grad=snllh, hess=ssnllh,
-            res=res, sres=sres
-        )
+        if mode == Objective.MODE_FUN:
+            return nllh, snllh, ssnllh
+        elif mode == Objective.MODE_RES:
+            return res, sres
```

Comment:
why is this here? i think fun, i.e. here _call_amici, should return just the values as a tuple that were asked for in tuple form, so e.g. fval if sensi_orders is (0,), or (fval, grad) if (0,1). now always 3 resp. 2 values are returned, looks like this migt lead to errors when using e.g. derivative-free optimizers. does this really work? haven't checked, can do that when i have time.

Reply:
`_call_amici` implements `fun` and `res` which are supposed to return the outputs exactly in that order. The subindexing according to the specified `sensi_orders` is applied in `_call_mode_fun` and `_call_mode_res` and `map_to_output`.